packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -180,15 +180,15 @@ export class MonacoOutlineContribution implements FrontendApplicationContributio
             (sameEndLine && endColSmallerOrEqual || endLineSmaller)) || !rangeBased);
     }
 
-    protected getRangeFromSymbolInformation(symbolInformation: DocumentSymbol): Range {
+    protected getRangeFromSymbolInformation(documentSymbol: DocumentSymbol): Range {
```

Comment:
Thanks for the remark. I was not careful enough and did not check other places.

It opens a question: why do we need that complex `parentContains` at all? As I see you have already switched to the `DocumentSymbol` from `SymbolInformation`. Why cannot we just use `!!parent.children && parent.children.indexOf(symbol) >= 0;` instead ?

From the [spec](https://microsoft.github.io/language-server-protocol/specification#textDocument_documentSymbol):
_The range doesn't have to denote a node range in the sense of a abstract syntax tree. It can therefore not be used to re-construct a hierarchy of the symbols._

The new specification supports the hierarchies, we can get rid of the hierarchy inferencing code, cannot we?

Also, if we could have an `m2p` converter here, we could get rid of the entire method:
 - use the `symbol.selectionRange` to select the name of the symbol in the editor,
 - use `symbol.range` to check if we are in the range of an element when changing the selection in the editor.

By the way, [does it work from the editor to the outline view](https://github.com/theia-ide/theia/issues/2883)? It seems to be "broken", and you are much familiar with the code-base. If it is a missing feature, I try to cover it withing this PR. Thanks!

Reply:
Because some language server still can return old symbol informations, `monaco-languageclient` converts them to document symbols without children and we should apply the old logic for such symbols.

Reply:
Regarding #2883 - I'm not sure whether it used to work before, but sounds good.

Reply:
Sorry, I have to disagree. Then this code, what we have here then should belong to the `monaco-languageclient` and not here:
```
        const symbolRange = this.getRangeFromSymbolInformation(symbol);
        const nodeRange = this.getRangeFromSymbolInformation(parent);
        const sameStartLine = symbolRange.start.line === nodeRange.start.line;
        const startColGreaterOrEqual = symbolRange.start.character >= nodeRange.start.character;
        const startLineGreater = symbolRange.start.line > nodeRange.start.line;
        const sameEndLine = symbolRange.end.line === nodeRange.end.line;
        const endColSmallerOrEqual = symbolRange.end.character <= nodeRange.end.character;
        const endLineSmaller = symbolRange.end.line < nodeRange.end.line;
        return (((sameStartLine && startColGreaterOrEqual || startLineGreater) &&
            (sameEndLine && endColSmallerOrEqual || endLineSmaller)) || !rangeBased);
```
It makes no sense maintain two diefferent logic in See:  `hierarchicalDocumentSymbolSupport`.

Reply:
Let me know if it is better in the `monaco-languageclient`, and I can create a PR.

Reply:
Yes, you're right. I will just fix the issue.

> By the way, [does it work from the editor to the outline view](https://github.com/theia-ide/theia/issues/2883)? It seems to be "broken", and you are much familiar with the code-base. If it is a missing feature, I try to cover it withing this PR. Thanks!

Is it a missing feature or a bug. You could save me some time... Thanks!

Reply:
https://github.com/theia-ide/theia/pull/2885#discussion_r217799827